beacon-chain/blockchain/receive_block.go

```diff
@@ -88,13 +96,15 @@ func (s *Service) ReceiveBlock(ctx context.Context, block interfaces.ReadOnlySig
 	if err := eg.Wait(); err != nil {
 		return err
 	}
+	if err := s.isDataAvailable(ctx, blockRoot, blockCopy); err != nil {
```

Comment:
The goroutine running this will block until the other blobs for the block appear via gossip, or the slot deadline hits. I think this blocking behavior before the fc lock is just as intended. cc @potuz